mst/mst.go

```diff
@@ -270,35 +439,113 @@ func (mst *MerkleSearchTree) Add(ctx context.Context, key string, val cid.Cid, k
 		}
 
 		checkTreeInvariant(updated)
-		newRoot, err := create(ctx, mst.cst, updated, keyZeros, mst.fanout)
-		if err != nil {
-			return nil, fmt.Errorf("creating new tree after split: %w", err)
-		}
+		newRoot := NewMST(mst.cst, cid.Undef, updated, keyZeros)
 
-		// why invalidate?
+		// we have mutated this node without recomputing CID, so invalidate it
```

Comment:
its technically a no-op, we set validPtr to be whether or not the cid we give it is defined, and here we're giving it an explicitly undefined cid